test/integration/scheduler/taint_test.go

```diff
@@ -198,116 +228,142 @@ func TestTaintNodeByCondition(t *testing.T) {
+			nodeConditions: []v1.NodeCondition{
 				{
-					Name:  "busybox",
-					Image: "busybox",
+					Type:   v1.NodePIDPressure,
+					Status: v1.ConditionTrue,
+				},
+				{
+					Type:   v1.NodeMemoryPressure,
+					Status: v1.ConditionTrue,
+				},
+				{
+					Type:   v1.NodeDiskPressure,
+					Status: v1.ConditionTrue,
+				},
+				{
+					Type:   v1.NodeReady,
+					Status: v1.ConditionTrue,
 				},
 			},
-			Tolerations: []v1.Toleration{
+			expectedTaints: []v1.Taint{
 				{
-					Key:      algorithm.TaintNodeNetworkUnavailable,
-					Operator: v1.TolerationOpExists,
-					Effect:   v1.TaintEffectNoSchedule,
+					Key:    algorithm.TaintNodeDiskPressure,
+					Effect: v1.TaintEffectNoSchedule,
+				},
+				{
+					Key:    algorithm.TaintNodeMemoryPressure,
+					Effect: v1.TaintEffectNoSchedule,
+				},
+				{
+					Key:    algorithm.TaintNodePIDPressure,
+					Effect: v1.TaintEffectNoSchedule,
 				},
 			},
 		},
 	}
 
-	if _, err := clientset.CoreV1().Pods(nsName).Create(networkDaemonPod); err != nil {
-		t.Errorf("Case 4: Failed to create pod for network daemon: %v", err)
-	} else {
-		if err := waitForPodToScheduleWithTimeout(clientset, networkDaemonPod, time.Second*60); err != nil {
-			t.Errorf("Case 4: Failed to schedule network daemon pod in 60s.")
-		}
-	}
-
-	// Case 5: Taint node by unschedulable condition
-	unschedulableNode := &v1.Node{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "node-2",
-		},
-		Spec: v1.NodeSpec{
-			Unschedulable: true,
-		},
-		Status: v1.NodeStatus{
-			Capacity: v1.ResourceList{
-				v1.ResourceCPU:    resource.MustParse("4000m"),
-				v1.ResourceMemory: resource.MustParse("16Gi"),
-				v1.ResourcePods:   resource.MustParse("110"),
+	for i, test := range tests {
+		node := &v1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-1",
 			},
-			Allocatable: v1.ResourceList{
-				v1.ResourceCPU:    resource.MustParse("4000m"),
-				v1.ResourceMemory: resource.MustParse("16Gi"),
-				v1.ResourcePods:   resource.MustParse("110"),
+			Spec: v1.NodeSpec{
+				Unschedulable: test.unschedulable,
 			},
-		},
-	}
+			Status: v1.NodeStatus{
+				Capacity:    nodeRes,
+				Allocatable: nodeRes,
+				Conditions:  test.nodeConditions,
+			},
+		}
+
+		if _, err := cs.CoreV1().Nodes().Create(node); err != nil {
+			t.Errorf("Case %d (%s): Failed to create node, err: %v", i, test.name, err)
+		}
 
-	nodeInformerCh2 := make(chan bool)
-	nodeInformer2 := informers.Core().V1().Nodes().Informer()
-	nodeInformer2.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		UpdateFunc: func(old, cur interface{}) {
-			curNode := cur.(*v1.Node)
-			if curNode.Name != "node-2" {
-				return
+		if err := waitForNodeTaints(cs, node, test.expectedTaints); err != nil {
+			t.Errorf("Case %d (%s): Failed to taint node, err: %v", i, test.name, err)
+		}
+
+		var pods []*v1.Pod
+
+		for j, p := range test.pods {
+			pod := p.pod.DeepCopy()
+			pod.Name = fmt.Sprintf("%s-%d", pod.Name, j)
+			pod.Spec.Tolerations = p.tolerations
```

Comment:
Not sure where my previous comment went, but I dont any need for deepcopy and then assigning name and tolerations, you could create the test pod that already contains name and tolerations to avoid the above.

Reply:
Give a copy to avoid unexpected reused data :), and try to avoid duplicated pod declare in test case list.

Reply:
I am not convinced that you need a deepcopy here.

Reply:
Do you mean just remove the deepcopy? What's your concern to the deepcopy? The performance is ok for UT, as most Pods (garantee, best-effort) are reused by pointer, that'll be safer to avoid unexpected reuse by other developer.